libdnf/rpm/reldep.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright (C) 2018-2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "libdnf/rpm/reldep.hpp"
+#include "sack_impl.hpp"
+#include "solv/reldep_parser.hpp"
+
+/* workaround, libsolv lacks 'extern "C"' in its header file */
+extern "C" {
+#include <solv/pool.h>
+#include <solv/pool_parserpmrichdep.h>
+#include <solv/util.h>
+}
+
+#include <stdexcept>
+
+namespace libdnf::rpm {
+
+
+Reldep::Reldep(Sack * sack, ReldepId id)
+        : sack(sack)
+        , id(id)
+{}
+
+Reldep::Reldep(Sack * sack, const char * name, const char * version, CmpType cmp_type)
+        : sack(sack)
+{
+    id = get_reldep_id(sack, name, version, cmp_type);
+}
+
+Reldep::Reldep(Sack * sack, const std::string & reldep_string)
+        : sack(sack)
+{
+    id = get_reldep_id(sack, reldep_string);
+}
+
+
+Reldep::Reldep(const Reldep & reldep)
+        : sack(reldep.sack)
+        , id(reldep.id)
+{}
+
+Reldep::Reldep(const Reldep && reldep) noexcept
+        : sack(reldep.sack)
+        , id(reldep.id)
+{}
+
+Reldep::~Reldep() = default;
+const char * Reldep::get_name() const { return pool_id2str(sack->pImpl->pool, id.id); }
+const char * Reldep::get_relation() const { return pool_id2rel(sack->pImpl->pool, id.id); }
+const char * Reldep::get_version() const { return pool_id2evr(sack->pImpl->pool, id.id); }
+const char * Reldep::to_string() const { return pool_dep2str(sack->pImpl->pool, id.id); }
+
+ReldepId
```

Comment:
Style: Put the return type on the same line as method name. Try clang-format. The same repeats on lot of places.

Reply:
Fixed by clang-format